source/D2Common/src/D2StatList.cpp

```diff
@@ -2782,117 +1833,73 @@ uint32_t* __stdcall D2COMMON_STATES_GetListGfxFlags_6FDB8AC0(D2UnitStrc* pUnit)
 }
 
 //D2Common.0x6FDB8B10 (#10516)
-void __stdcall D2Common_10516(D2UnitStrc* pUnit, int nFrame)
+void __stdcall STATLIST_UpdateStatListsExpiration(D2UnitStrc* pUnit, int nFrame)
 {
-	D2StatListStrc* pStatListEx = NULL;
-	D2StatListStrc* pPrevious = NULL;
-
-	if (pUnit && pUnit->pStatListEx && (pUnit->pStatListEx->dwFlags & 0x80000000) != 0)
+	if (pUnit && pUnit->pStatListEx && STATLIST_IsExtended(pUnit->pStatListEx))
 	{
-		pStatListEx = pUnit->pStatListEx->pMyLastList;
-		if (nFrame)
+		if (D2StatListStrc* pStatListEx = pUnit->pStatListEx->pMyLastList)
 		{
-			if (pStatListEx)
+			D2StatListStrc* pPrevious = nullptr;
+			do
 			{
-				do
+				pPrevious = pStatListEx->pPrevLink;
+				if (nFrame == 0)
 				{
-					pPrevious = pStatListEx->pPrevLink;
-					if (pStatListEx->dwFlags & 2 && (int)pStatListEx->dwExpireFrame <= nFrame)
-					{
-						if (!(pStatListEx->dwFlags & 0x80000000))
-						{
-							D2Common_STATLIST_FreeStatListEx_6FDB7050((D2StatListExStrc*)pStatListEx);
-						}
-
-						pPrevious = pUnit->pStatListEx->pMyLastList;
-					}
-					pStatListEx = pPrevious;
+					pStatListEx->dwExpireFrame--;
```

Comment:
In the original code, this is only done, if (pStatListEx->dwFlags & STATLIST_NEWLENGTH) is true